lib/ansible/module_utils/oneview.py

```diff
@@ -0,0 +1,782 @@
+
+        Returns:
+            The resource found or None.
+        """
+        result = self.resource_client.get_by('name', name)
+        return result[0] if result else None
+
+    def resource_present(self, resource, fact_name, create_method='create'):
+        """
+        Generic implementation of the present state for the OneView resources.
+        It checks if the resource needs to be created or updated.
+
+        Args:
+            resource (dict):
+                Resource to create or update.
+            fact_name (str):
+                Name of the fact returned to the Ansible.
+            create_method (str):
+                Function of the OneView client that will be called for resource creation. Usually create or add.
+
+        Returns:
+            A dictionary with the expected arguments for the AnsibleModule.exit_json
+        """
+
+        changed = False
+        if "newName" in self.data:
+            self.data["name"] = self.data.pop("newName")
+
+        if not resource:
+            resource = getattr(self.resource_client, create_method)(self.data)
+            msg = self.MSG_CREATED
+            changed = True
+
+        else:
+            merged_data = resource.copy()
+            merged_data.update(self.data)
+
+            if ResourceComparator.compare(resource, merged_data):
+                msg = self.MSG_ALREADY_PRESENT
+            else:
+                resource = self.resource_client.update(merged_data)
+                changed = True
+                msg = self.MSG_UPDATED
+
+        return dict(
+            msg=msg,
+            changed=changed,
+            ansible_facts={fact_name: resource}
+        )
+
+    @staticmethod
+    def transform_list_to_dict(list_):
+        """
+        Transforms a list into a dictionary, putting values as keys.
+
+        Args:
+            list_: List of values
+
+        Returns:
+            dict: dictionary built
+        """
+
+        ret = {}
+
+        if not list_:
+            return ret
+
+        for value in list_:
+            if isinstance(value, dict):
+                ret.update(value)
+            else:
+                ret[str(value)] = True
+
+        return ret
+
+    @staticmethod
+    def get_logger(mod_name):
+        """
+        To activate logs, setup the environment var LOGFILE
+        e.g.: export LOGFILE=/tmp/ansible-oneview.log
+
+        Args:
+            mod_name: module name
+
+        Returns: Logger instance
+        """
+
+        logger = logging.getLogger(os.path.basename(mod_name))
+        global LOGFILE
+        LOGFILE = os.environ.get('LOGFILE')
+        if not LOGFILE:
+            logger.addHandler(logging.NullHandler())
+        else:
+            logging.basicConfig(level=logging.DEBUG, datefmt='%Y-%m-%d %H:%M:%S',
+                                format='%(asctime)s %(levelname)s %(name)s %(message)s',
+                                filename=LOGFILE, filemode='a')
+        return logger
```

Comment:
@bcoca what do you think of this?  Should the modules just use AnsibleModule.log() instead?